client_plugin/drivers/shared/etcdops.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	etcdClient "github.com/coreos/etcd/clientv3"
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/swarm"
+)
+
+const (
+	apiVersion               = "v1.24" // docker engine 1.24 and above support this api version
+	dockerUSocket            = "unix:///var/run/docker.sock"
+	etcdClientPort           = ":2379"
+	etcdPeerPort             = ":2380"
+	etcdClusterToken         = "vsphere-shared-etcd-cluster"
+	etcdListenURL            = "0.0.0.0"
+	etcdScheme               = "http://"
+	etcdClusterStateNew      = "new"
+	etcdClusterStateExisting = "existing"
+)
+
+// InitEtcd start or join ETCD cluster depending on the role of the node
+func InitEtcd() error {
+	// create new docker client
```

Comment:
It will depend on how and where we will use docker client.
I am fine to put it into the VolumeDriver structure later when it's used in other situations.

I am not sure about getsDockerInfo() function though.
The returned info already has those information, why bother setting them into another struct?
And those information are changing so we cannot reuse them...